packages/object/addon/index.js

```diff
@@ -276,17 +279,50 @@ export const wrapComputed = computedDecoratorWithParams((desc, params) => {
   @function
   @param {...String} propertyNames - Names of the properties that trigger the function
  */
+const CHAINS_FINISHED = new WeakMap();
+
 export const observes = decoratorWithRequiredParams((desc, params) => {
   assert(
     'The @observes decorator must be applied to functions',
     desc && desc.descriptor && typeof desc.descriptor.value === 'function'
   );
 
+  // hackity hackity haaaaaack
+  desc.extras = [
+    {
+      kind: 'field',
+      placement: 'own',
+      key: '__EMBER_DECORATORS_FINISH_CHAINS__',
+      descriptor: {
+        enumerable: false,
+        writable: true,
+        configurable: true,
+      },
+      initializer() {
+        if (!CHAINS_FINISHED.has(this)) {
+          Ember.finishChains(Ember.meta(this));
```

Comment:
Hmm, we _don't_ need to do this if working on `Ember.Object`'s though (AFAICT). Can we avoid the double call to `.finishChains` when working on Ember.Object's?

Reply:
Yes, definitely, I actually had that a little while ago, must not have committed for some reason...